src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -37,37 +42,101 @@
 
   ParameterizedTypeName(ClassName rawType, List<TypeName> typeArguments,
       List<AnnotationSpec> annotations) {
+    this(
+        rawType,
+        rawType.enclosingClassName() != null ? rawType.enclosingClassName() : rawType,
```

Comment:
Sorry about that; originally I completely removed rawType; but because it was public, there were things accessing it, so I had to add it back.  Updated to use rawType & cleaned up a bit around it.